samples/chat-demo/app.go

```diff
@@ -32,13 +37,41 @@ type ChatApp struct {
 	// The only state of the application is the chat message history,
 	// to which each delivered request appends one message.
 	messages []string
+
+	// The current epoch number.
+	currentEpoch t.EpochNr
+
+	// For each epoch number, stores the corresponding membership.
+	memberships []map[t.NodeID]t.NodeAddress
+
+	// Network transport module used by Mir.
+	// The app needs a reference to it in order to manage connections when the membership changes.
+	transport net.Transport
```

Comment:
I believe that we have to use a sync primitive here when accessing  `currentEpoch` and `memberships`. I found that in my implementation of the state manager in Eudico.

Reply:
Now I completely removed `currentEpoch` (the chat app does not need to keep track of it any more) and `memberships` is never accessed concurrently (unless I missed something). So I think we should be fine without synchronizing (at least for the chat app - might be different for the Eudico integration)